lib/db/memory.js

```diff
@@ -194,40 +210,67 @@ MemoryStore.prototype = {
     return P.resolve(_code);
   },
   getCode: function getCode(code) {
-    return P.resolve(this.codes[unbuf(encrypt.hash(code))]);
+    return P.resolve(clone(this.codes[unbuf(encrypt.hash(code))]));
   },
   removeCode: function removeCode(id) {
     delete this.codes[unbuf(id)];
     return P.resolve();
   },
-  generateToken: function generateToken(vals) {
-    var token = {};
-    token.clientId = vals.clientId;
-    token.userId = vals.userId;
-    token.email = vals.email;
-    token.scope = vals.scope;
-    token.createdAt = new Date();
-    token.type = 'bearer';
-    var _token = unique.token();
-    var ret = clone(token);
-    token.token = encrypt.hash(_token);
-    this.tokens[unbuf(token.token)] = token;
-    ret.token = _token;
+  generateAccessToken: function generateAccessToken(vals) {
+    var token = unique.token();
+    var now = new Date();
+    var t = {
+      clientId: vals.clientId,
+      userId: vals.userId,
+      email: vals.email,
+      scope: vals.scope,
+      type: 'bearer',
+      createdAt: now,
+      // ttl is in seconds
+      expiresAt: new Date(+now + (vals.ttl || MAX_TTL) * 1000),
+      token: encrypt.hash(token)
+    };
+    var ret = clone(t);
+    this.tokens[unbuf(t.token)] = t;
+    ret.token = token;
     return P.resolve(ret);
   },
-  getToken: function getToken(token) {
-    return P.resolve(this.tokens[unbuf(token)]);
+  getAccessToken: function getAccessToken(token) {
+    return P.resolve(clone(this.tokens[unbuf(token)]));
   },
-  removeToken: function removeToken(id) {
+  removeAccessToken: function removeAccessToken(id) {
     delete this.tokens[unbuf(id)];
     return P.resolve();
   },
+  generateRefreshToken: function generateRefreshToken(vals) {
+    var token = unique.token();
+    var t = {
+      clientId: vals.clientId,
+      userId: vals.userId,
+      email: vals.email,
+      scope: vals.scope,
+      createdAt: new Date(),
+      token: encrypt.hash(token)
```

Comment:
nit: it may be clearer to name this e.g. `tokenId` or `tokenHash` rather than just `token`